### ISSUE TITLE
Error clearing current chat and Error: could not delete item

### ISSUE THREAD

**Issue by wchucgf** [microsoft/sample-app-aoai-chatGPT#244]
when clicking "broom" icon to clear chat window i get "Error clearing current chat"
and
when trying to delete a chat history item or all chat history i get "Error: could not delete item"

any idea why it might be?
- the cosmos key i provided is the read+write key...

**Comment by wchucgf** [comment]
"DELETE /history/delete HTTP/1.1" 500 error - when "broom" clear button is clicked....

**Comment by edercarlima** [comment]
I'm also having the same problem. I downloaded it from github and deployed it.
I ask the questions and when I click to clear, it doesn't clear.
To enable cleaning and storage of history, do you need to perform some procedure or upload the json file that is in the infrastructure through the azure template?

**Comment by wchucgf** [comment]
> @wchucgf @edercarlima Hi, does this issue still persist? If you cloned the repo and deployed manually, I would ensure that you have the following environment variables in your web app or local env:
> 
> ![image](https://user-images.githubusercontent.com/43946287/272360375-66eafb42-3412-477f-8023-2b2bdda4a6c4.png)

yes issue persists. correct i cloned the repo and deployed to my already existing web apps and cosmos. those variables are set else the history wouldn't get logged - history gets logged fine, just cant delete it any of the history

**Comment by edercarlima** [comment]
@wchucgf I managed to solve the problem through a correction made through another issue.

**Comment by AmirTamj** [comment]
> @wchucgf I managed to solve the problem through a correction made through another issue.

@edercarlima Any pointer or steps that you could share to resolve this?

**Comment by robertfalcone** [comment]
How did you resolve the issue?

**Comment by jasonhumsft** [comment]
Sharing how we resolved similar issue here. 

In our case, the "conversations" container in Cosmos DB is created manually, and initially the container was created with wrong partition key "/userid", but the code writes all data with property name "userId" (and the code doesn't use "userid" at all, so the value of "userid" property for all data is empty). 

When the code tries to delete data with the value of "userId" as partition key, it couldn't locate the partition and raises 404 error. It was resolved by recreating the "conversations" container with correct partition key "/userId".

If anyone encounters similar issue, maybe can double check how the "conversations" container was created. If possible, using the Bicep templates included in this project to create all resources.
